smartfile/__init__.py

```diff
@@ -124,8 +124,40 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        try:
+            return self.post('/path/oper/remove', path=deletefile)
+        except KeyError:
+            raise Exception("Destination file does not exist")
+        
+    def upload(self, usrfile):
+        # needed to split the tuple
+        newtuple = usrfile
+        # splits tuple to allow ability to remove "/" at the end, if present 
+        ourfile, newdata = newtuple
+        if ourfile[-1:] == "/":
```

Comment:
I would remove the tuple in favor of accepting two separate parameters. The code above is quite confusing. I don't know that copying the tuple is necessary, you can index into a tuple. Also, the comment is wrong since the "/" is not removed, instead an Exception is raised.

`if usrfile[0].endswith('/'):`